org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/core/manipulation/StubUtility.java

```diff
@@ -1146,23 +1148,30 @@ private static void add(String[] names, Set<String> result) {
 	private static String getBaseNameFromExpression(IJavaProject project, Expression assignedExpression, int variableKind) {
 		String name= null;
 		if (assignedExpression instanceof CastExpression) {
-			assignedExpression= ((CastExpression)assignedExpression).getExpression();
+			assignedExpression= ((CastExpression) assignedExpression).getExpression();
 		}
 		if (assignedExpression instanceof Name) {
-			Name simpleNode= (Name)assignedExpression;
+			Name simpleNode= (Name) assignedExpression;
 			IBinding binding= simpleNode.resolveBinding();
 			if (binding instanceof IVariableBinding)
-				return getBaseName((IVariableBinding)binding, project);
+				return getBaseName((IVariableBinding) binding, project);
 
 			return ASTNodes.getSimpleNameIdentifier(simpleNode);
 		} else if (assignedExpression instanceof MethodInvocation) {
-			name= ((MethodInvocation)assignedExpression).getName().getIdentifier();
+			MethodInvocation methodInvocation= (MethodInvocation) assignedExpression;
+			name= methodInvocation.getName().getIdentifier();
+			if (name.equals("next")) { //$NON-NLS-1$
+				Expression receiver= methodInvocation.getExpression();
+				String modifiedName= getBaseNameFromReceiver(receiver);
+				if (!modifiedName.equals("element")) //$NON-NLS-1$
+					return modifiedName;
+			}
 		} else if (assignedExpression instanceof SuperMethodInvocation) {
-			name= ((SuperMethodInvocation)assignedExpression).getName().getIdentifier();
+			name= ((SuperMethodInvocation) assignedExpression).getName().getIdentifier();
```

Comment:
Something in your format profile is adding a blank between casts and their element to cast.  This should not be changed as part of this patch.  Other than that, the code looks good.  If you want me to fix it with my own formatting, I can do so.

Reply:
Thanks a lot! I will fix that immediately.